staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/equality.go

```diff
@@ -50,9 +50,12 @@ var (
 	avoidNoopTimestampUpdatesEnabled = determineAvoidNoopTimestampUpdatesEnabled()
 )
 
```

Comment:
an alternative could be to initialize this package var once on first use

```
var (
  avoidTimestampEqualities conversion.Equalities
  initAvoidTimestampEqualities sync.Once
)
func getAvoidTimestampEqualities() conversion.Equalities {
  initAvoidTimestampEqualities.Do(func(){
    var eqs = equality.Semantic.Copy()
    ...
  }())
}
```

then access it exclusively through `getAvoidTimestampEqualities()`

that lets everything that is contributing to `equality.Semantic` in init() blocks work as is

Reply:
I've updated the code hopefully matching what you had in mind